### PR TITLE
Test Fix: libtester test failures in CICD

### PR DESCRIPTION
**Pull request by oschwaldp-oci**
Resolves: https://github.com/AntelopeIO/leap/issues/1437

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -255,7 +255,8 @@ jobs:
           token: ${{github.token}}
       - name: Install cdt Packages
         run: |
-          apt install -y ./*.deb
+          apt-get update
```
Comment (spoonincode) [inline .github/workflows/build.yaml:258]: there is already another `apt-get update` in the same job earlier, but only for the `deb-install` case
https://github.com/AntelopeIO/leap/blob/9ff0583b33259b965bd11f3bc228cd2a30e7c814/.github/workflows/build.yaml#L236-L239
Could consider a single step near the beginning of the job that always does the `apt-get update`, to avoid situations where it's done twice in the same job, and avoid the duplication.

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -91,8 +91,8 @@ jobs:
           cpack
       - name: Install dev package
         run: |
-          apt update && apt upgrade -y
-          apt install -y ./build/leap_*.deb ./build/leap-dev*.deb
+          apt-get update
```
Comment (spoonincode) [inline .github/workflows/build.yaml:94]: reason for removal of `upgrade`? there was some uncertainly around best practices last time this came up. Not doing the `upgrade` means you could end up with "mixed date" dependency installations: if `foo` depends on `bar`, and `bar` is already installed _from a year ago_, doing `apt update && apt install foo` I believe means the latest version of `foo` will be using a one year old `bar` (unless `foo` has an explicit version dependency)

But, I believe Ubuntu tries to maintain ABI compatibility within a release? That seems potentially hard over the course of 4 years. But maybe.

TLDR it was never clear to me the canonical best practice here; but always doing a `upgrade` seems like the safest option.

**Review by spoonincode — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 1 above)
Comment (oschwaldp-oci) [reply .github/workflows/build.yaml]: Good idea. Done.  Only implemented a new step in the `libtester-tests` job as the other looked currently to be and continue to be a single use scenario.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (oschwaldp-oci) [reply .github/workflows/build.yaml]: Yeah it is difficult to say. Your comment had me go refresh on this as well.  My reasoning here was that the `upgrade` step can force one's hand into installing a host of updated packages that aren't being used by our use case, whereas relying on the deb package install to take care of meeting its own dependency tree would be safe -- so long as the package index had been updated.  Anyway, for now I've added it back in and we can punt this down the road for future discussion...or not.

**Review by heifner — APPROVED** [review]
